storage/innobase/xtrabackup/src/xtrabackup.cc

```diff
@@ -5116,15 +5115,15 @@ static bool xtrabackup_init_temp_log(void) {
     goto error;
   }
 
-  log_format = (Log_format)mach_read_from_4(log_buf + LOG_HEADER_FORMAT);
-  xtrabackup_original_log_format = log_format;
+  log_format =
+      static_cast<Log_format>(mach_read_from_4(log_buf + LOG_HEADER_FORMAT));
+  log_detected_format = log_format;
   if (log_format == Log_format::VERSION_8_0_28 ||
       log_format == Log_format::VERSION_8_0_19) {
     // We interpret v4 & v5 as v6 by writing proper start lsn header later in
     // this function
     log_format = Log_format::VERSION_8_0_30;
   }
-  log_detected_format = log_format;
```

Comment:
few lines below we have this line:
```
if (log_format < Log_format::VERSION_8_0_1) {
```
This becomes invalid if we use log_format. Because it is saved as v6 always.